packages/editor/src/components/entities-saved-states/entity-record-item.js

```diff
@@ -28,6 +28,30 @@ export default function EntityRecordItem( {
 		return parents[ parents.length - 1 ];
 	}, [] );
 
+	// Handle templates that might use default descriptive titles
+	const entityRecordTitle = useSelect(
+		( select ) => {
+			if (
+				! select( 'core/edit-site' ) ||
+				'postType' !== kind ||
+				'wp_template' !== name
+			) {
+				return title;
+			}
+
+			const template = select( 'core' ).getEditedEntityRecord(
+				kind,
+				name,
+				key
+			);
+			const { title: templateTitle } = select(
+				'core/edit-site'
+			).getTemplateInfo( template );
+			return templateTitle;
+		},
+		[ name, kind, title, key ]
+	);
```

Comment:
This is an awkward place to use the `core/edit-site` store, as it might be used outside of the Site Editor, reason why we check for the existence of `select( 'core/edit-site' )` right at the beginning of it.

The intention is that when we edit templates, we want to see its correct title (as in: the one expected by the user, consistent with other places in the UI) in the edits review panel.
Since `auto-draft` entities don't have a title, editing for the first time a theme-provided template in the Site Editor might result in displaying an incorrect title (undefined, or the slug, etc.).

Reply:
> This is an awkward place to use the core/edit-site store

Would it make more sense to have the templateInfo store in `core/editor` instead of `core/edit-site`?  While it is most relevant to the site editor, it is valuable information regardless.  Similarly, it seems ok to access `core/editor` store in edit-site a bit more than it does to access `core/edit-site` in editor. 🤔

Reply:
This sounds like a great idea actually!
`core/editor`'s `editorSettings` is already full of very specific stuff, I guess adding default template types in there too wouldn't do much harm. 😅

Reply:
@Addison-Stavlo After looking into it, I'm deciding against it.
The reason is that `edit-post` and `edit-site` initialize the editor in wildly different ways, and they add their PHP-provided settings to different stores.
I don't know the reason behind this, and what kind of inconsistencies it is currently creating.
Either way, I'm not going to attempt a major refactor of the editors initialization in this PR. 😄 

The `select( 'core/edit-site' )` is slightly smelly, but it works for our purposes.
As far as I could test, the entity review seems to work as expected in `edit-post`, displaying the correct template title.
By checking for `core/edit-site`, we are also preventing regressions in other editors that we might not catch.